Framework/CurveFitting/test/Algorithms/FitTest.h

```diff
@@ -538,6 +541,259 @@ class FitTest : public CxxTest::TestSuite {
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000, 1.000000,
+        1.000000,
+    };
+
+    X.back() = X[98] + dx;
+    AnalysisDataService::Instance().add("ResolutionTest_WS", ws);
+
+    Algorithms::Fit fit;
+    fit.initialize();
+
+    fit.setPropertyValue(
+        "Function", "composite=Convolution,"
+                    "FixResolution=true,NumDeriv=true;name=Resolution,FileName="
+                    "\"ResolutionTestResolution.res\","
+                    "WorkspaceIndex=0;name=ResolutionTest_Gauss,c=5,h=2,s=1");
+    fit.setPropertyValue("InputWorkspace", "ResolutionTest_WS");
+    fit.setPropertyValue("WorkspaceIndex", "0");
+    fit.execute();
+  }
+
+  void getStretchExpMockData(Mantid::MantidVec &y, Mantid::MantidVec &e) {
+    // values extracted from y(x)=2*exp(-(x/4)^0.5)
+    y[0] = 2;
+    y[1] = 1.2130613;
+    y[2] = 0.98613738;
+    y[3] = 0.84124005;
+    y[4] = 0.73575888;
+    y[5] = 0.65384379;
+    y[6] = 0.58766531;
+    y[7] = 0.53273643;
+    y[8] = 0.48623347;
+    y[9] = 0.44626032;
+    y[10] = 0.41148132;
+    y[11] = 0.38092026;
+    y[12] = 0.35384241;
+    y[13] = 0.32968143;
+    y[14] = 0.30799199;
+    y[15] = 0.28841799;
+    y[16] = 0.27067057;
+    y[17] = 0.25451242;
+    y[18] = 0.2397465;
+    y[19] = 0.22620756;
+
+    for (int i = 0; i <= 19; i++) {
+      // estimate errors as ten percent of the "count number"
+      const double cc = 0.1;
+      e[i] = cc * y[i];
+    }
+  }
+
+  void test_function_StretchExp_Against_MockData() {
+    Algorithms::Fit alg2;
+    TS_ASSERT_THROWS_NOTHING(alg2.initialize());
+    TS_ASSERT(alg2.isInitialized());
+
+    // create mock data to test against
+    std::string wsName = "StretchExpMockData";
+    int histogramNumber = 1;
+    int timechannels = 20;
+    Workspace_sptr ws = WorkspaceFactory::Instance().create(
+        "Workspace2D", histogramNumber, timechannels, timechannels);
+    Workspace2D_sptr ws2D = boost::dynamic_pointer_cast<Workspace2D>(ws);
+    // in this case, x-values are just the running index
+    for (int i = 0; i < 20; i++)
+      ws2D->dataX(0)[i] = 1.0 * i + 0.00001;
```

Comment:
In general we want to obtain a reference to `dataX(0)` before entering the loop.